scene/gui/tree.cpp

```diff
@@ -544,6 +544,21 @@ bool TreeItem::is_collapsed() {
 	return collapsed;
 }
 
+void TreeItem::set_visible(bool p_visible) {
+	if (visible == p_visible) {
+		return;
+	}
+	visible = p_visible;
+	if (tree) {
+		tree->update();
```

Comment:
This will update the tree even if a parent item is invisible (so needlessly). I wonder if it's worth to optimize it 🤔

Reply:
I am not sure if there is a way to do this without checking parents upwards until one is found to be invisible.
I am guessing that this would still be more efficient than updating the entire tree each time.
If you think it's worth including this in this PR I can do that otherwise we can leave for a future one?

Reply:
I had some further thought on this and realised I don't think that the `set_collapsed` method has an optimisation like you described above either (so if you collapse or un-collapse a node which is underneath an already collapsed node you would toggle an update of the whole tree.
Maybe it might be worth doing an investigation on some optimisations in a separate ticket/PR.